### PR TITLE
Admissions: Refactor Area of Study college field to use taxonomy

### LINKED ISSUE #3223: Admissions - AoS College Reference Field limited to just Colleges, not all basic pages
# Story

As an editor when filling out the AoS I only want want to see auto results for Colleges, not all basic pages on the website.

# Proposed solution

- [x] Refactor as a taxonomy vocabulary "Colleges"
- [x] Add fields to vocabulary that allows Admissions to use the taxonomy as a replacement
- [x] Use a layout builder template for term rendering
- [x] Update the AoS content type to reference the taxonomy
- [ ] Update the render as necessary

### PR DESCRIPTION
**Pull request by pyrello**
Resolves #3223 

# How to test

* `blt ds --site=admissions.uiowa.edu`
* `drush @admissions.local uli admin/structure/taxonomy/manage/colleges/add` and create an entry for Tippie College of Business ([the original, for reference](https://admissions.uiowa.edu/colleges/tippie-college-business)).
* Visit http://admissions.local.drupal.uiowa.edu/node/1021/edit and add "Tippie College of Business" in the "College(s)" field.
* Save the page and observe that the link shows up in the "Related links" section.
* Click the link to the term page. Observe that "Business" shows up underneath the college information.

### CODE REVIEW

**Comment by pyrello** [comment]
The unexpected discovery of this PR is that the work done in #3193 was limiting terms other than Academic groups from being viewable because the view was originally set up to validate against that vocabulary and show page not found for everything else. I experimented with trying to set up multiple displays to validate against different vocabularies, to allow for term pages to look different, but that doesn't work.

I found this module, which may allow for setting the behavior per vocabulary: https://www.drupal.org/project/tvi

Based on the current setup without that module, if we want to override default behavior with LB + views, we have to accept an all or nothing approach where all vocabularies are handled the same way or none of them are.

I think it would also be possible to not use views or LB and just handle overriding the default behavior by using template overrides and possibly some preprocessing. Not totally sure about whether this would work.

**Comment by pyrello** [comment]
I think this is all ready except for tweaking the way it looks. @bspeare if you want to add any finishing touches, feel free :)

**Comment by bspeare** [comment]
Nothing to add for now.   I was able to test it out and verify that the testing instructions worked as described.

**Comment by pyrello** [comment]
I'm going to wrap this up understanding that there will probably be some requests for visual updates.

**Comment by joewhitsitt** [comment]
pulling down to test

**Comment by bspeare** [comment]
A few items that we might want/need to add to this;
- "Areas of Study" heading above areas of study.
- [x] Change Areas of study list to be bulleted list items in a two column grid?

**Comment by joewhitsitt** [comment]
>"Areas of Study" heading above areas of study.

The headline will become available for all admissions views blocks once https://github.com/uiowa/uiowa/pull/3243 is merged. There is a @/todo comment to wire the headline up with the views title if available.

TIL there is a github user todo that links up if we mention a todo comment

**Comment by joewhitsitt** [comment]
- The flow of the form seems off with website below the published checkbox
- Webmaster doesn't have access to the taxonomy. Will webmaster have access to change this setting on a per term basis?
Screen Shot 2021-03-16 at 10 58 50 AM

File: docroot/themes/custom/uids_base/templates/field/field.html.twig [hunk 1]
```diff
@@ -69,7 +69,7 @@
       </div>
     {% else %}
         {% for item in items %}
-          <{{ item_tag }}{{ attributes.addClass(classes, 'field__item') }}>{{ item.content }}</{{ item_tag }}>
+          <{{ item_tag }}{{ attributes.addClass(classes, 'field__item') }}>{{ block('field_icon') }}{{ item.content }}</{{ item_tag }}>
```
Comment (joewhitsitt) [inline docroot/themes/custom/uids_base/templates/field/field.html.twig:72]: What does this do?

Comment (pyrello) [reply docroot/themes/custom/uids_base/templates/field/field.html.twig:72]: It allows us to use the `field_icon` block to be used when the label is hidden. That syntax is necessary when placing a twig `block` statement in a template multiple times.

**Comment by pyrello** [comment]
> The flow of the form seems off with website below the published checkbox

Since this is going to be a pretty limited use case for the time being and only set up for 11 terms, I wasn't too concerned with the niceties of giving the form the platform level treatment.

> Webmaster doesn't have access to the taxonomy. Will webmaster have access to change this setting on a per term basis?

This setting should not be configurable by anyone but developers.

**Comment by joewhitsitt** [comment]
>Webmaster doesn't have access to the taxonomy

Scott can't add/edit the terms?

**Comment by pyrello** [comment]
> > Webmaster doesn't have access to the taxonomy

> Scott can't add/edit the terms?

I didn't take a look at permissions. I will look at that.

**Comment by pyrello** [comment]
Added permissions. Thanks for catching that @joewhitsitt! That seems to be one of my regular blindspots.

**Comment by pyrello** [comment]
> The flow of the form seems off with website below the published checkbox

I moved the website field up the form. I also removed the TVI from displaying (it only displayed for Admins anyways) but we don't really need it per term.

**Comment by richardbporter** [comment]
I'm ready to migrate this! 👍🏻

**Comment by pyrello** [comment]
This is ready for review again.

**Comment by joewhitsitt** [comment]
pulling down again

**Review by joewhitsitt — APPROVED** [review]
Thanks for the changes, though I am confused about some others. 🤷

**Comment by pyrello** [comment]
> Thanks for the changes, though I am confused about some others. 🤷

What changes are you confused about?

**Comment by joewhitsitt** [comment]
Creating the separate view display and the title above the view when the headline field will soon be available.

**Pull request merged**